[docs] replay cli docs (#23024)
MystenLabs/sui: docs/content/references/cli/replay.mdx

---

> title: Sui Replay CLI

Reviewer: ```suggestion
title: Sui Replay CLI
description: Use the Sui CLI to replay a given transaction to view its effects.
---
```

---

Replay is a tool that allows you to replay a given transaction to view transaction effects. In addition, you can enable tracing to generate traces that can be used

> for debugging via the VSCode Move Plugin.

Reviewer: ```suggestion
for debugging via the VS Code Move plugin.
```

> The main command for replaying a transaction is shown below. For extra flags and arguments, run `sui client replay-transaction --help`.

Reviewer: ```suggestion
The main command for replaying a transaction follows. For all available flags and arguments, run `sui client replay-transaction --help`.
```

```
sui client replay-transaction --tx-digest <digest>
```

> After running such a command, the tool generates a directory corresponding to the digest in the current directory: `<cur_directory>/.replay/<digest>`, unless the `--output-dir` flag is set.

Reviewer: ```suggestion
After running this command, the tool generates a directory corresponding to the digest in the current directory: `<cur_directory>/.replay/<digest>`, unless you set the `--output-dir` flag.
```

> A typical response after running the command above is:

Reviewer: ```suggestion
After successfully running the previous command, your console provides a response similar to the following:
```

```
Successfully replayed transaction

> ╭───────────────────────────────────────────────────────────────────────────────────────────────────╮

Reviewer: As an aside: I'm thinking of changing all the output example to JSON. Do you hate that idea?

Author: I think that would be too verbose. We could maybe have it as side to side, in a different little tab?
